pydantic_ai_slim/pydantic_ai/toolsets/mapped.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from pydantic_core import SchemaValidator
+
+from .._run_context import AgentDepsT, RunContext
+from ..tools import (
+    ToolDefinition,
+)
+from . import AbstractToolset
+from .run import RunToolset
+from .wrapper import WrapperToolset
+
+
+@dataclass(init=False)
+class MappedToolset(WrapperToolset[AgentDepsT]):
+    """A toolset that maps the names of the tools it contains."""
```

Comment:
I feel this docstring doesn't adequately explain what this toolset does / in what circumstances it is intended to be used. Maybe you'll flesh that out as part of subsequent docs work, but just pointing it out. (I'm not confident about its purpose.)